fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -5496,6 +5496,24 @@ public void modifyTableDefaultReplicationNum(Database db, OlapTable table, Map<S
                 properties.get(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM));
     }
 
+    // The caller need to hold the table write lock
+    public void modifyTableDefaultBucketNum(Database db, OlapTable table, Map<String, String> properties) {
```

Comment:
Just table that not belongs to a colocate group and hash partitioned can change its bucket num.